laravel/app/Models/Assignment.php

```diff
@@ -19,4 +19,29 @@ class Assignment extends Model
      * @var bool
      */
     public $timestamps = false;
+
+    public function residents()
+    {
+        return $this->belongsTo(Resident::class);
+    }
+
+    public function options()
+    {
+        return $this->belongsTo(Option::class);
+    }
+
+    public function scheduledatas()
```

Comment:
should be singular `schedule_data` (or should it be `scheduleData`? I'm actually not sure what the correct naming convention is here), since an assignment only belongs to one ScheduleData.